### DISCUSSION TITLE
Is it possible access thiea's workspace in glsp server?

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1118] [Q&A]
hi,
   I use glsp node-server in my theia application. I want to access the opened workspace, What I know is by using theia workspaceService,
but It seems that can not directly access the workspaceService in the node server. Does anyone know what to do?

**Comment by planger** [comment]
Sorry for the late reply, your question somehow fell through the cracks!

As the node glsp server runs in a separate process it doesn't have direct access to the Theia services, such as the workspace service. The idea behind this separation is that your glsp server can be used across different tool platforms, such as VS Code, Eclipse Theia, Eclipse RCP, etc.

However, typically you access the workspace from your glsp server directly via the file API. Your GLSP client will send the file URI (or any custom parameter that you may add) to the server for initialization. Based on that, you know on the server what to load directly from the file system.

If you really need to know what the actual workspace location (and not just the opened file) is in your glsp server, you'd have to send this as a custom parameter to the glsp server.
